python/lsst/multiprofit/sourceconfig.py

```diff
@@ -0,0 +1,385 @@
+                component_configs[f"{prefix_group}{name_comp}"] = component_config
+        return component_configs
+
+    def get_integral_label_default(self) -> str:
+        prefix = "mix: ${name_group} " if self.has_prefix_group() else ""
+        return f"{prefix}{ComponentGroupConfig.get_integral_label_default()}"
+
+    def has_prefix_group(self) -> bool:
+        return (len(self.component_groups) > 1) or next(iter(self.component_groups.keys()))
+
+    def make_source(
+        self,
+        component_group_fluxes: list[list[Fluxes]],
+        label_integral: str | None = None,
+    ) -> [g2f.Source, list[g2f.Prior]]:
+        """Make a gauss2d.fit.Source from this configuration.
+
+        Parameters
+        ----------
+        component_group_fluxes
+            A list of Fluxes for each of the self.component_groups to use
+            when calling make_components.
+        label_integral
+            A label to apply to integral parameters. Can reference the
+            relevant component mixture name with ${name_group}.
+
+        Returns
+        -------
+        source
+            An appropriate gauss2d.fit.Source.
+        priors
+            A list of priors from all constituent components.
+        """
+        if label_integral is None:
+            label_integral = self.get_integral_label_default()
+        components, priors = self._make_components_priors(
+            component_group_fluxes=component_group_fluxes,
+            label_integral=label_integral,
+        )
+        source = g2f.Source(components)
+        return source, priors
+
+    def make_psf_model(
+        self,
+        component_group_fluxes: list[list[Fluxes]],
+        label_integral: str | None = None,
+    ) -> [g2f.PsfModel, list[g2f.Prior]]:
+        """Make a gauss2d.fit.PsfModel from this configuration.
+
+        This method will validate that the arguments make a valid PSF model,
+        i.e. with a unity total flux, and only one config for the none band.
+
+        Parameters
+        ----------
+        component_group_fluxes
+            A list of CentroidFluxes for each of the self.component_groups
+            when calling make_components.
+        label_integral
+            A label to apply to integral parameters. Can reference the
+            relevant component mixture name with ${name_group}.
+
+        Returns
+        -------
+        psf_model
+            An appropriate gauss2d.fit.PSfModel.
+        priors
+            A list of priors from all constituent components.
+        """
+        if label_integral is None:
+            label_integral = f"PSF {self.get_integral_label_default()}"
+        components, priors = self._make_components_priors(
+            component_group_fluxes=component_group_fluxes,
+            label_integral=label_integral,
+            validate_psf=True,
+        )
+        model = g2f.PsfModel(components=components)
+
+        return model, priors
```

Comment:
`make_source` and `make_psf_model` are nearly identical except that they build different objects from the components. This is just a suggestion, but I would recommend creating a `make_source_components` method that returns the components and priors and call that from both methods to avoid the duplicate code.

Reply:
There is already a `_make_components_priors` method. Did you mean add a public version of that? I'm not sure what purpose that would serve.

This could be consolidated into a single method but I think in this case I anticipated the potential for future divergence and decided to live with the duplication.

Reply:
Ok, if you see them diverging more in the future that's fine. It's just that they're both 9 lines of code, and they share 7 of those lines, so it seemed like a place to consolidate.